.buildkite/commands/install-node-dependencies.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -eu
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+
+CACHEKEY="$BUILDKITE_PIPELINE_SLUG-npm-$PLATFORM-$ARCHITECTURE-node-$NODE_VERSION-$PACKAGE_HASH"
+
+LOCAL_NPM_CACHE=./vendor/npm
+mkdir -p $LOCAL_NPM_CACHE
+echo "--- :npm: Set npm to use $LOCAL_NPM_CACHE for cache"
+npm set cache $LOCAL_NPM_CACHE
+echo "npm cache set to $(npm get cache)"
```

Comment:
In setting this up, it seems like using a local cache works better than the default (`$HOME/.npm`).

I haven't gone back and did a fair comparison, but I got the impression that the `_cacache` subfolder was not extracted when using the local cache.

Reply:
I am no `npm` expert so I don't know all the available commands nor how it handles cache, nor what `npm ci` does that `npm install` wouldn't, but couldn't [our `install_npm_dependencies` helper from ci-toolkit](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/blob/trunk/bin/install_npm_packages) be of help here (instead of reinventing the wheel)?
And if not, could we maybe extend `a8c-ci-toolkit` to bring the things from this script that are not yet supported by our helper to be supported (either by the existing `install_npm_packages` helper, or by a new one)? 🤔